### PR TITLE
chore: make lance-arrow compilable for wasm target

### PR DESCRIPTION
**Pull request by eddyxu**

### CODE REVIEW

File: rust/lance-arrow/Cargo.toml [hunk 1]
```diff
@@ -10,6 +10,9 @@ description = "Arrow Extension for Lance"
 keywords.workspace = true
 categories.workspace = true
 
+[lib]
+crate-type = ["cdylib", "rlib"]
```
Comment (wjones127) [inline rust/lance-arrow/Cargo.toml:14]: Why does this need to be `cdylib`?

Comment (eddyxu) [reply rust/lance-arrow/Cargo.toml:14]: This is required for `wasm-pack`

Comment (wjones127) [reply rust/lance-arrow/Cargo.toml:14]: Okay. It's just hard to tell from this PR. In the future, consider adding more comments as well as a CI job to continue to verify WASM actually works with the crate.

**Review by wjones127 — APPROVED** [review]

File: rust/lance-arrow/Cargo.toml (hunk 1 above)
Comment (eddyxu) [reply rust/lance-arrow/Cargo.toml:14]: Sg. I will setup a wasm CI this weekend if i can find some bandwidth.

**Pull request merged**